acme/src/certificate.rs

```diff
@@ -49,13 +61,24 @@ impl RustyAcme {
 
                 // only verify that leaf has the right identity fields
                 if i == 0 {
-                    Self::verify_leaf_certificate(cert, &order.try_get_coalesce_identifier()?, hash_alg, env)?;
+                    Self::verify_leaf_certificate(cert, &order.try_get_coalesce_identifier()?, hash_alg, env.as_ref())?;
```

Comment:
```suggestion
                    Self::verify_leaf_certificate(cert, &order.try_get_coalesce_identifier()?, hash_alg, &env)?;
```

Reply:
This was not interchangeable since env is an `Option<PkiEnvironment>`

`env.as.ref()` = `Option<&PkiEnvironment>`
`&env` = `&Option<PkiEnvironment>`

Reply:
oh. my bad. I completely misread the code.